lib/storage/providers/MemoryOnlyProvider.ts

```diff
@@ -0,0 +1,144 @@
+    init() {
+        // do nothing
+    },
+
+    /**
+     * Get the value of a given key or return `null` if it's not available in memory
+     */
+    getItem(key) {
+        const value = store[key];
+
+        return Promise.resolve(value === undefined ? null : value);
+    },
+
+    /**
+     * Get multiple key-value pairs for the give array of keys in a batch.
+     */
+    multiGet(keys) {
+        const getPromises = _.map(keys, (key) => new Promise((resolve) => this.getItem(key).then((value) => resolve([key, value])))) as Array<Promise<KeyValuePair>>;
+        return Promise.all(getPromises);
+    },
+
+    /**
+     * Sets the value for a given key. The only requirement is that the value should be serializable to JSON string
+     */
+    setItem(key, value) {
+        set(key, value);
+
+        return Promise.resolve();
+    },
+
+    /**
+     * Stores multiple key-value pairs in a batch
+     */
+    multiSet(pairs) {
+        const setPromises = _.map(pairs, ([key, value]) => this.setItem(key, value));
+        return new Promise((resolve) => Promise.all(setPromises).then(() => resolve()));
+    },
+
+    /**
+     * Merging an existing value with a new one
+     */
+    mergeItem(key, _changes, modifiedData) {
+        // Since Onyx already merged the existing value with the changes, we can just set the value directly
+        return this.setItem(key, modifiedData);
+    },
+
+    /**
+     * Multiple merging of existing and new values in a batch
+     * This function also removes all nested null values from an object.
+     */
+    multiMerge(pairs) {
+        _.forEach(pairs, ([key, value]) => {
+            const existingValue = store[key] as unknown as Record<string, unknown>;
+            const newValue = utils.fastMerge(existingValue, value as unknown as Record<string, unknown>) as unknown as Value;
+
+            set(key, newValue);
+        });
+
+        return Promise.resolve([]);
+    },
+
+    /**
+     * Remove given key and it's value from memory
+     */
+    removeItem(key) {
+        delete store[key];
+        return Promise.resolve();
+    },
+
+    /**
+     * Remove given keys and their values from memory
+     */
+    removeItems(keys) {
+        _.each(keys, (key) => {
+            delete store[key];
+        });
+        return Promise.resolve();
+    },
+
+    /**
+     * Clear everything from memory
+     */
+    clear() {
+        store = {};
+        return Promise.resolve();
+    },
+
+    // This is a noop for now in order to keep clients from crashing see https://github.com/Expensify/Expensify/issues/312438
+    setMemoryOnlyKeys() {
+        // do nothing
+    },
+
+    /**
+     * Returns all keys available in memory
+     */
+    getAllKeys() {
+        return Promise.resolve(_.keys(store));
+    },
+
+    /**
+     * Gets the total bytes of the store.
+     * `bytesRemaining` will always be `Number.POSITIVE_INFINITY` since we don't have a hard limit on memory.
+     */
+    getDatabaseSize() {
```

Comment:
why do we need to measure the database size?

Reply:
We added this for logging purposes. It only gets used when we encounter an error so we probably don't need this. In theory, the `MemoryOnlyProvider` should never fail or throw any errors (at least doesn't look like it can).

Reply:
See: https://github.com/Expensify/react-native-onyx/blob/69b2d5973b4aef19ce783440df55dedb30d4efd2/lib/Onyx.js#L1037-L1045

And: https://github.com/Expensify/react-native-onyx/blob/69b2d5973b4aef19ce783440df55dedb30d4efd2/lib/Onyx.js#L1069-L1078

Reply:
ah I can see from the code that it's only used in a very rare case. Thank you so much for the explanation!